mmv1/third_party/terraform/services/storage/resource_storage_bucket_object.go

```diff
@@ -513,3 +556,36 @@ func expandCustomerEncryption(input []interface{}) map[string]string {
 	}
 	return expanded
 }
+
+func expandObjectRetention(configured interface{}) *storage.ObjectRetention {
+	retentions := configured.([]interface{})
+	if len(retentions) == 0 {
+		return nil
+	}
+	retention := retentions[0].(map[string]interface{})
+
+	mode := retention["mode"].(string)
+
+	objectRetention := &storage.ObjectRetention{
+		RetainUntilTime: retention["retain_until_time"].(string),
+		Mode:            mode,
```

Comment:
nit: no longer need to set this separately.

```suggestion
		Mode:            retention["mode"].(string),
```